docs/write_functions.adoc

```diff
@@ -0,0 +1,118 @@
+
+=== Write Functions
+
+Each field and register is parameterized with a _write function_. By default,
+the write function is `groov::w::replace`.
+
+[source,cpp]
+----
+using my_field = groov::field<"field", std::uint32_t, 3, 0, groov::w::replace>;
+----
+
+==== Write functions
+
+Built-in write functions are in the `groov::w` namespace. Examples are:
+
+- `replace` ("normal write")
+- `read_only`
+
+There are 16 possible write functions, each characterized by a truth table
+relating the current value *C*, the written value *W* and the result *R*. For
+example:
+
+.Truth table for replacement
+[width=30%,cols="1,1,1"]
+|===
+^h|C ^h|W ^h|R
+
+^|0 ^|0 ^|0
+^|0 ^|1 ^|1
+^|1 ^|0 ^|0
+^|1 ^|1 ^|1
+|===
+
+.Truth table for read-only
+[width=30%,cols="1,1,1"]
+|===
+^h|C ^h|W ^h|R
+
+^|0 ^|0 ^|0
+^|0 ^|1 ^|0
+^|1 ^|0 ^|1
+^|1 ^|1 ^|1
+|===
+
+==== Identity specs
+
+Each write function may have an identity spec indicated by its truth table and
+exposed as the alias `id_spec`. For replacement, the identity spec is `none`.
+For read-only, the identity spec is `any`. There are 4 possible identity specs:
+
+- `none`
+- `zero`
+- `one`
+- `any`
+
+Each identity spec (except `none`) has a function that computes the identity for
+a field according to its bit range:
+
+[source,cpp]
+----
+template <std::unsigned_integral T, std::size_t Msb, std::size_t Lsb>
+constexpr static auto identity() -> T;
+----
+
+==== Building the write masks
+
+In general a given write to a register will only want to write some of the field
+values. The other field values are left unspecified. The write machinery uses
+the field write functions and their associated identity specs to compute:
+
+- the write mask for the fields to be written
+- the identity mask for the other fields
+- the identity value (if any) for the other fields
+
+For example:
+
+[source,cpp]
+----
+// the default write function is groov::w::replace
+using my_field_0 = groov::field<"field_0", std::uint32_t, 15, 0>;
+using my_field_1 = groov::field<"field_1", std::uint32_t, 31, 16>;
```

Comment:
suggestion: you might want to make this unaligned to avoid confusion with potential subword write optimizations.